lib/iris/coords.py

```diff
@@ -55,7 +55,8 @@
 class CoordDefn(namedtuple('CoordDefn',
                            ['standard_name', 'long_name',
                             'var_name', 'units',
-                            'attributes', 'coord_system'])):
+                            'attributes', 'coord_system',
+                            'bounds_are_climatological'])):
```

Comment:
@trexfeathers A small thing, but a big thing...

The implication of supporting the CF `climatology` attribute simply means that the `bounds` are climatological - that's it.

Therefore, I don't think that this needs to be explicitly qualified here. Would you consider changing `bounds_are_climatological` to simply `climatology`?

If `climatology` is `True`, therefore the `bounds` are *special*, otherwise not. I find it a hard sell to force users to use `bounds_are_climatological`

Reply:
`bounds_are_climatological` is a mouthful I agree. But I am worried about a word that could have too broad implications, such as `climatology`. For me the greatest importance is clarity on the limited scope of the property, and so that the name we use needs to begin with `bounds_are_`. I suggest `bounds_are_climatology`, which should be a _little_ less clunky than the current name.

@abooton please could you relay what you have discussed with me, so that we can get every side of this recorded in the PR?

Reply:
Okay, how about I throw I throw this curve ball into the mix...

Reference [Appendix A: Attrributes](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#attribute-appendix) in **Table A.1. Attributes**, the `climatology` attribute has the **Description** of `Identifies a climatology variable`.

So call the property `climatology_variable` because according to CF, that's what it is :wink:

Reply:
Actually, re-reading the CF conventions again, the climatology variable is the actual cube that the time axis coordinate is attached to, where the time axis is deemed a climatological time axis defined within a climatological time coordinate variable (well, from my understanding).

That said, the following statement from the CF conventions [7.4. Climatological Statistics](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#climatological-statistics) is relevant here:

> _"... climatological time coordinate variable does not have a `bounds` attribute. Instead, it has a `climatology` attribute ..."_

The difficulty here is that we're in the business of implementing the conventions and we're overloading (for convenience and the common sense balance between effort and benefit) the use of the `bounds` property, and need a simple means to discriminate it's overloaded use between _normal_ and _climatological_ time coordinates.

In the pure CF conventions world `bounds` and `climatology` as controlled attributes are mutually exclusive, so you could also argue that introducing a property of `climatology` is really incorrect here. However, having a property of `climatological` (as much as I hate typing that word) make sense in this context.

Given that we're advocating that the data model of `iris` is based on the CF conventions (with a healthy peppering of pragmatism due to conventions vs implementations dilemmas) I believe it's fine to borrow from the CF conventions vernacular without being accused of being file format specific. 

That said, and given the discussion so far, I'd recommend the use of `climatological` as a property name, but on the proviso that this can **only** be set on a **time coordinate**, which implies enforcing that the `iris.Coord.units.is_time_reference()` is checked **before** setting the `climatological` property to `True`. Otherwise, it's not valid to set the `climatological` property on just any old coordinate.

Reply:
Sounds ideal @bjlittle I'll get on this now

Reply:
I have two thoughts on this...
1) I think it would be clearer from a user perspective to have the flag called `climatological_bounds`.  (The use of "climatology" confuses me) I usually use the term ‘climatology’ to refer to the data phenomenon averaged over a number of years.  Everything else is "climatological" as it is associated with the climatology.  Incidentally JULES has a "is_climatological" flag for this.
2) What behaviour are we after?  From the users perspective I think using a "bounds_are_climatological" flag rather than something akin to "is_climatological" indicates different behaviour, in that the former I expect the bound info will be reset on save, the later hints that on save the entire coord and associated attributes will be "climatological".  i.e. the cell_methods name would get updated upon save too.

Reply:
@abooton Thanks. 

We're simply going with `climatological` as a property name. 

It is a simple flag and only applies to a time coordinate, and is in keeping with the behaviour alluded to in the CF conventions [7.4. Climatological Statistics](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#climatological-statistics) i.e., this is a climatological time coordinate. The implementation implication for us is that the `bounds` of the coordinate are `climatological` - in this regard the use of `iris.coord.Coord.bounds` is overloaded and I deem the effort to subclass in this instance not worth the end benefit. We're playing the pragmatic card.

From a conventions purist perspective, this whole issue could be circumvented by having a `ClimatologicalCoord` sub-classed from `DimCoord`, which has a `climatology` rather than `bounds`, but the fall-out within `iris` would be unreasonable.

There may be further connective tissue with regards to enforcing certain behaviour for cell methods given a climatological time coordinate and performing climatological statistics, but for the sake of progress I'd argue that is outside the scope of this particular PR.

Reply:
`climatological` is good.  As long as the doc string accurately indicates the current behaviour I think it is fine. We will have a nicely named variable from which to extent the "climatological" scope in the future as required.